index.go

```diff
@@ -177,17 +261,48 @@ func (idx *faissIndex) SearchWithoutIDs(x []float32, k int64, exclude []int64, p
 		defer excludeSelector.Delete()
 	}
 
-	searchParams, err := NewSearchParams(idx, params, selector)
+	tempParamsBytes := make([]byte, 0)
```

Comment:
Don't do any allocs here - line 269 takes care of it. 
The idea is to minimise allocs wherever possible.